Build failure: epsonscan2

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

- update system flake to e.g. rev `9abb87b552b7f55ac8916b6fc9e5cb486656a2f3`
- install `epsonscan2` in eg `environment.systemPackages`
- (I am not using overrides/overlays for this package)

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/286350298

### Relevant log output

```console
Running phase: qtPreHook
Running phase: unpackPhase
unpacking source archive /nix/store/ls0s7w2268rxm4nmlllkpqfj2048ldb8-source
source root is source
Running phase: patchPhase
applying patch /nix/store/jzd2w2cwsb65jzyfzqndrm1vpj27sylw-build.patch
patching file CMakeLists.txt
patching file src/CMakeLists.txt
patching file src/Controller/Src/Filter/GetOrientation.cpp
Hunk #1 succeeded at 27 with fuzz 1 (offset 15 lines).
patching file src/Controller/Src/Scanner/Engine.cpp
patching file src/Standalone/CMakeLists.txt
applying patch /nix/store/4js7vns38g561zn4i7v2v0fkk9p7hjfh-gcc14.patch
patching file thirdparty/zlib/gzlib.c
patching file thirdparty/zlib/gzread.c
patching file thirdparty/zlib/gzwrite.c
patch unexpectedly ends in middle of line
patch unexpectedly ends in middle of line
applying patch /nix/store/qm18x9grmwsa67zzmjsj7nglsjlw0al8-0002-Fix-crash.patch
patching file src/ES2Command/Src/Interface/ipc/ipcInterfaceImpl.cpp
Hunk #1 succeeded at 129 (offset 1 line).
applying patch /nix/store/k0hnqn5mrm2qnmigxq128k35bx468wap-0004-Fix-a-crash-on-an-OOB-container-access.patch
patching file src/Controller/Src/Finder/Finder.cpp
applying patch /nix/store/fgb3abwcz7pkxfpvbqy6ahw0d89n2agn-0003-Use-XDG-open-to-open-the-directory.patch
patching file src/Standalone/savedestfolder.cpp
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/share/doc/epsonscan2 -DCMAKE_INSTALL_INFODIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/strip -DCMAKE_RANLIB=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/ranlib -DCMAKE_AR=/nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0 -DCMAKE_INSTALL_LIBDIR=lib/x86_64-linux-gnu
CMake Warning (dev) at CMakeLists.txt:19 (project):
  cmake_minimum_required() should be called prior to this top-level project()
  call.  Please see the cmake-commands(7) manual for usage documentation of
  both commands.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/591apldfgm1gr238rbfc8ib0y1z8ng2y-gcc-wrapper-14-20241116/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:20 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Warning (dev) at CMakeLists.txt:51 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /nix/store/jkwsbly2wga09y64hnx5sckql6vwqhsj-boost-1.87.0-dev/lib/cmake/Boost-1.87.0/BoostConfig.cmake (found suitable version "1.87.0", minimum required is "1.36.0")
 Boost_INCLUDE_DIRS: /nix/store/jkwsbly2wga09y64hnx5sckql6vwqhsj-boost-1.87.0-dev/include
 Boost_LIBRARIES: 
 Boost_LIBRARY_DIRS: 
-- Found JPEG: /nix/store/4zq9mzi2g4ld46rsxn2h0s562ij29vyc-libjpeg-turbo-3.0.4/lib/libjpeg.so (found version "62")
-- Found ZLIB: /nix/store/zv9shv0566mrq0lvrws448rs52gc56k1-zlib-1.3.1/lib/libz.so (found version "1.3.1")
-- Found PNG: /nix/store/m04yfi9gd23mjlwsarcwz955zbz4fgix-libpng-apng-1.6.43/lib/libpng.so (found version "1.6.43")
-- Found TIFF: /nix/store/7g7nsc9570gmz85w0fliycwshl2l8bcw-libtiff-4.7.0-dev/lib/cmake/tiff (found version "4.7.0")
 JPEG_INCLUDE_DIRS: /nix/store/d9l9hixc9aviw4xriwk4xykycgr57wci-libjpeg-turbo-3.0.4-dev/include
 JPEG_LIBRARY_DIRS: /nix/store/4zq9mzi2g4ld46rsxn2h0s562ij29vyc-libjpeg-turbo-3.0.4/lib/libjpeg.so
TIFF INCLUDE : /nix/store/7g7nsc9570gmz85w0fliycwshl2l8bcw-libtiff-4.7.0-dev/include
TIFF LIBRARY : optimized;/nix/store/n8z46r30s3a5d9647pxbcadf3mnzw12g-libtiff-4.7.0/lib/libtiff.so.6.1.0;debug;/nix/store/n8z46r30s3a5d9647pxbcadf3mnzw12g-libtiff-4.7.0/lib/libtiff.so.6.1.0
PNG INCLUDE : /nix/store/5rvyn65jr9qcy1gcsz91zd3j5c65xjij-libpng-apng-1.6.43-dev/include;/nix/store/qr0p57jasfpkcacypjsyi62f69vyad3q-zlib-1.3.1-dev/include
PNG LIBRARY : /nix/store/m04yfi9gd23mjlwsarcwz955zbz4fgix-libpng-apng-1.6.43/lib/libpng.so;/nix/store/zv9shv0566mrq0lvrws448rs52gc56k1-zlib-1.3.1/lib/libz.so
ZLIB INCLUDE : /nix/store/qr0p57jasfpkcacypjsyi62f69vyad3q-zlib-1.3.1-dev/include
ZLIB LIBRARY : /nix/store/zv9shv0566mrq0lvrws448rs52gc56k1-zlib-1.3.1/lib/libz.so
EPSONSCAN_PATH epsonscan2/
COMMON_ETC_PATH /nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/etc
EPSON_WORK_PATH /tmp/epsonWork/
EPSON_SETTINGS_PATH /homeless-shelter/.epsonscan2/
EPSON_INSTALL_PATH /nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/lib/x86_64-linux-gnu/epsonscan2/
COMMON_SHARE_PATH /nix/store/va7iwncak2d7m63d4i5z9yjwzf9lgv62-epsonscan2-6.7.70.0/share
Library architecture x86_64
CMake Deprecation Warning at src/CMakeLists.txt:19 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Deprecation Warning at src/Standalone/CMakeLists.txt:20 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Warning (dev) at /nix/store/rriq96lra5fa2fm91q9c2zdswfk05pqi-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:44 (message):
  qt5_use_modules is not part of the official API, and might be removed in Qt
  6.
Call Stack (most recent call first):
  /nix/store/rriq96lra5fa2fm91q9c2zdswfk05pqi-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:434 (_qt5_warn_deprecated)
  src/Standalone/CMakeLists.txt:169 (QT5_USE_MODULES)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at src/DetectAlert/CMakeLists.txt:20 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


CMake Warning (dev) at /nix/store/rriq96lra5fa2fm91q9c2zdswfk05pqi-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:44 (message):
  qt5_use_modules is not part of the official API, and might be removed in Qt
  6.
Call Stack (most recent call first):
  /nix/store/rriq96lra5fa2fm91q9c2zdswfk05pqi-qtbase-5.15.16-dev/lib/cmake/Qt5Core/Qt5CoreMacros.cmake:434 (_qt5_warn_deprecated)
  src/DetectAlert/CMakeLists.txt:64 (QT5_USE_MODULES)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Deprecation Warning at src/ScanSDK/CMakeLists.txt:20 (cmake_minimum_required):
  Compatibility with CMake < 3.10 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
  to tell CMake that the project requires at least <min> but has been updated
  to work with policies introduced by <max> or earlier.


-- Configuring done (2.5s)
CMake Warning (dev) in src/Standalone/CMakeLists.txt:
  Policy CMP0043 is not set: Ignore COMPILE_DEFINITIONS_<Config> properties.
  Run "cmake --help-policy CMP0043" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/DetectAlert/CMakeLists.txt:
  Policy CMP0043 is not set: Ignore COMPILE_DEFINITIONS_<Config> properties.
  Run "cmake --help-policy CMP0043" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/Standalone/CMakeLists.txt:
  Policy CMP0043 is not set: Ignore COMPILE_DEFINITIONS_<Config> properties.
  Run "cmake --help-policy CMP0043" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/DetectAlert/CMakeLists.txt:
  Policy CMP0043 is not set: Ignore COMPILE_DEFINITIONS_<Config> properties.
  Run "cmake --help-policy CMP0043" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done (0.3s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
build flags: -j2 SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash
[  0%] Building CXX object src/CommonUtility/Linux/CMakeFiles/commonutility.dir/__/__/Platform.cpp.o
[  0%] Building CXX object src/CommonUtility/Linux/CMakeFiles/commonutility.dir/__/utils/ESJsonUtils.cpp.o
In file included from include/boost/none.hpp:17,
                 from include/boost/optional/optional.hpp:53,
                 from include/boost/optional.hpp:15,
                 from include/boost/format/internals.hpp:21,
                 from include/boost/format.hpp:38,
                 from /build/source/src/CommonUtility/Linux/../../Platform.h:38,
                 from /build/source/src/CommonUtility/utils/Json/../../ESCommonTypedef.h:27,
                 from /build/source/src/CommonUtility/utils/Json/../ESJsonUtils.h:21,
                 from /build/source/src/CommonUtility/utils/Json/JsonCommon.h:28,
                 from /build/source/src/CommonUtility/utils/ESJsonUtils.cpp:20:
include/boost/none_t.hpp:24:2: error: #error "Boost.Optional requires some C++11 features since version 1.87. If you have an older C++ version use Boost.Optional version 1.86 or earlier."
In file included from include/boost/none.hpp:17,
                 from include/boost/optional/optional.hpp:53,
                 from include/boost/optional.hpp:15,
                 from include/boost/format/internals.hpp:21,
                 from include/boost/format.hpp:38,
                 from /build/source/src/Platform.h:38,
                 from /build/source/src/Platform.cpp:19:
include/boost/none_t.hpp:24:2: error: #error "Boost.Optional requires some C++11 features since version 1.87. If you have an older C++ version use Boost.Optional version 1.86 or earlier."
/build/source/src/Platform.cpp: In function 'DWORD GetModuleFileName(LPTSTR, DWORD)':
/build/source/src/Platform.cpp:278:10: warning: ignoring return value of 'ssize_t readlink(const char*, char*, size_t)' declared with attribute 'warn_unused_result' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wno-unused-result-Wunused-result8;;]
  278 |  readlink( "/proc/self/exe", lpFilename, buf_size - 1 );
      |  ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [src/CommonUtility/Linux/CMakeFiles/commonutility.dir/build.make:79: src/CommonUtility/Linux/CMakeFiles/commonutility.dir/__/__/Platform.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [src/CommonUtility/Linux/CMakeFiles/commonutility.dir/build.make:93: src/CommonUtility/Linux/CMakeFiles/commonutility.dir/__/utils/ESJsonUtils.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:290: src/CommonUtility/Linux/CMakeFiles/commonutility.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.69, NixOS, 25.05 (Warbler), 25.05.20250104.8f3e1f8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/khbvilmsrv4l69nwd52h27j1mp44a0xi-source`


### Notify maintainers

@james-atkins

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ep/epsonscan2/package.nix
+++ b/pkgs/by-name/ep/epsonscan2/package.nix
@@ -70,7 +70,7 @@ stdenv.mkDerivation {
 
   postPatch = ''
     substituteInPlace src/Controller/Src/Scanner/Engine.cpp \
-      --replace-fail '@KILLALL@' ${killall}/bin/killall
+      --replace-fail '@KILLALL@' ${lib.getExe killall}
 
     substituteInPlace src/Controller/Src/Filter/GetOrientation.cpp \
       --replace-fail '@OCR_ENGINE_GETROTATE@' $out/libexec/epsonscan2-ocr/ocr-engine-getrotate
@@ -137,7 +137,7 @@ stdenv.mkDerivation {
 
   desktopItems = lib.optionals withGui [
     (makeDesktopItem {
-      name = pname;
+      name = "epsonscan2";
       exec = "epsonscan2";
       icon = "epsonscan2";
       desktopName = "Epson Scan 2";
@@ -154,22 +154,27 @@ stdenv.mkDerivation {
     inherit description;
     mainProgram = "epsonscan2";
     longDescription = ''
-      Epson Scan 2 scanner driver including optional non-free plugins such as OCR and network
-      scanning.
+      The Epson Scan 2 scanner driver, including optional non-free plugins such
+      as OCR and network scanning.
 
       To use the SANE backend:
-      <literal>
-      hardware.sane.extraBackends = [ pkgs.epsonscan2 ];
-      </literal>
+      ```nix
+      {
+        hardware.sane.extraBackends = [ pkgs.epsonscan2 ];
+      }
+      ```
 
-      Overrides can be used to customise this package. For example, to enable non-free plugins and
-      disable the Epson GUI:
-      <literal>
-      pkgs.epsonscan2.override { withNonFreePlugins = true; withGui = false; }
-      </literal>
+      Overrides can be used to customise this package. For example, to enable
+      non-free plugins and disable the Epson GUI:
+      ```nix
+      pkgs.epsonscan2.override {
+        withNonFreePlugins = true;
+        withGui = false;
+      }
+      ```
     '';
     homepage = "https://support.epson.net/linux/en/epsonscan2.php";
-    platforms = [ "x86_64-linux" ];
+    platforms = lib.systems.inspect.patternLogicalAnd lib.systems.inspect.patterns.isLinux lib.systems.inspect.patterns.isx86_64;
     sourceProvenance =
       with lib.sourceTypes;
       [ fromSource ] ++ lib.optionals withNonFreePlugins [ binaryNativeCode ];

--- a/pkgs/by-name/ep/epsonscan2/package.nix
+++ b/pkgs/by-name/ep/epsonscan2/package.nix
@@ -25,13 +25,17 @@
 let
   pname = "epsonscan2";
   description = "Epson Scan 2 scanner driver for many modern Epson scanners and multifunction printers";
-  version = "6.7.70.0";
+  # Epson updates projects without changing their version numbers.
+  # There can be multiple different packages identified by the same
+  #version, so we also tag them with the month and year shown in
+  # the Epson download page.
+  version = "6.7.70.0-01-2025";
 
   system = stdenv.hostPlatform.system;
 
   src = fetchzip {
-    url = "https://download3.ebz.epson.net/dsc/f/03/00/16/14/37/7577ee65efdad48ee2d2f38d9eda75418e490552/epsonscan2-6.7.70.0-1.src.tar.gz";
-    hash = "sha256-y7XGxrOpVou/ZSfUffV3qv+SsFFpTiU7pWvtfsiLZWc=";
+    url = "https://download3.ebz.epson.net/dsc/f/03/00/16/60/70/c7fc14e41ec84255008c6125b63bcac40f55e11c/epsonscan2-6.7.70.0-1.src.tar.gz";
+    hash = "sha256-WvNBy5hAxNJfwgjBR5VGaXuyTCK2YEiD3i7SMDl3U/U=";
   };
   bundle =
     {
@@ -65,8 +69,6 @@ stdenv.mkDerivation {
   ];
 
   postPatch = ''
-    rm CMakeCache.txt
-
     substituteInPlace src/Controller/Src/Scanner/Engine.cpp \
       --replace-fail '@KILLALL@' ${killall}/bin/killall
 

--- a/pkgs/by-name/ep/epsonscan2/package.nix
+++ b/pkgs/by-name/ep/epsonscan2/package.nix
@@ -2,7 +2,7 @@
   lib,
   stdenv,
   autoPatchelfHook,
-  boost,
+  boost186,
   cmake,
   copyDesktopItems,
   imagemagick,
@@ -90,7 +90,7 @@ stdenv.mkDerivation {
 
   buildInputs =
     [
-      boost
+      boost186 # uses Boost.Optional but epsonscan2 is pre-C++11.
       libjpeg
       libpng
       libtiff

